src/Auth0.ManagementApi/Clients/BaseClient.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Auth0.Core.Http;
+using System;
+using System.Collections.Generic;
+
+namespace Auth0.ManagementApi.Clients
+{
+    /// <summary>
+    /// Base class from which all Client classes inherit.
+    /// </summary>
+    public abstract class BaseClient
+    {
+        /// <summary>
+        /// <see cref="IManagementConnection"/> used to make all API calls.
+        /// </summary>
+        protected IManagementConnection Connection { get; }
+
+        /// <summary>
+        /// <see cref="Uri"/> of the endpoint to use in making API calls.
+        /// </summary>
+        protected Uri BaseUri { get; }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="BaseClient"/>.
+        /// </summary>
+        /// <param name="connection"><see cref="IManagementConnection"/> used to make all API calls.</param>
+        /// <param name="baseUri"><see cref="Uri"/> of the endpoint to use in making API calls.</param>
+        protected BaseClient(IManagementConnection connection, Uri baseUri)
```

Comment:
**Question:** This is a constructor, correct? It's interesting that this class can declare its constructor as `protected` but extending class constructors are public. Is that a .NET thing or am I missing something fundamental about what's going on here?

Reply:
It basically say yes, only subclasses can call this.  It's a good practice but doesn't actually make that much difference as the class is abstract so can't be created directly anyway.